src/components/external-source/ExternalSourcePanelEntry.svelte

```diff
@@ -0,0 +1,140 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import Eye from 'bootstrap-icons/icons/eye-fill.svg?component';
+  import EyeSlash from 'bootstrap-icons/icons/eye-slash.svg?component';
+  import { externalSources, planDerivationGroupLinks } from '../../stores/external-source';
+  import { plan } from '../../stores/plan';
+  import { plugins } from '../../stores/plugins';
+  import type { User } from '../../types/app';
+  import type { DerivationGroup, ExternalSourceSlim } from '../../types/external-source';
+  import effects from '../../utilities/effects';
+  import { formatDate } from '../../utilities/time';
+  import { tooltip } from '../../utilities/tooltip';
+  import Collapse from '../Collapse.svelte';
+
+  export let derivationGroup: DerivationGroup;
+  export let user: User | null;
+
+  let relevantSources: ExternalSourceSlim[] = [];
+
+  $: enabled = ($plan !== null && $planDerivationGroupLinks
+                                    .find(a => a.derivation_group_name === derivationGroup.name && a.plan_id === $plan.id)?.enabled) ?? true
+  $: relevantSources = $externalSources.filter(source => derivationGroup.name === source.pkey.derivation_group_name);
+
+  function onChange() {
+    // call effect
+    console.log("calling effect")
+    effects.updatePlanDerivationGroupEnabled(derivationGroup.name, $plan, !enabled, user);
+  }
+
+  async function deleteEmptyDerivationGroup() {
+    if (enabled) {
+      // Delete plan derivation group association
+      await effects.deleteDerivationGroupForPlan(derivationGroup.name, $plan, user);
+    }
+    // Delete the derivation group itself
+    await effects.deleteDerivationGroup(derivationGroup.name, user);
+  }
+</script>
+
+<div class="external-source-pairing">
+  <Collapse
+    title={derivationGroup.name}
+    tooltipContent={'Derivation group ' + derivationGroup.name}
+    defaultExpanded={false}
+  >
+    <span slot="right" style:display="flex">
+      <p class="derived-event-text">
+        {derivationGroup.derived_event_total} derived events
+      </p>
+      {#if enabled === true}
+        <button
+          class="st-button icon"
+          on:click|stopPropagation={onChange}
+          use:tooltip={{ content: 'Show in timeline', placement: 'top' }}
```

Comment:
Confused about the behavior here – seems like it is actually controlling whether this external event is included in the plan and not just temporarily hiding it on the timeline? If so this should really be some sort of "remove"? Or you could implement a different sort of visibility state that hides it on the timeline without removing it from the plan. Feels weird when I hide this from the timeline and refresh the page and the derivation group has disappeared.

Reply:
This "hide" is implemented more like its an extension of the 'view' definition itself - so you aren't _de-linking_ it from the plan (i.e., in the future if you had constraints around this external event, they would still be successful if the event was hidden this way), but when you hide it that hide will stay implemented until that event/filter is specifically _un-hidden_. Up to discussion on this though, if it feels more natural to you to have that filter be temporary and removed on things such as, de-linking & re-linking a derivation group that was previously hidden, and navigating away from that page.

Reply:
Suggestion is to make it ephemeral, much like how constraints and hiding are implemented. It will _not_ touch the `enabled` field in the `PlanDerivationGroup` table.

Reply:
Addressed in [updated hiding DGs to just operate locally like constraints do](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/024cf90db585267f827c4b50019ffc41ba8abd6c). Also updated in aerie main - removed enabled field (see [remove enabled field from plan_derivation_group](https://github.com/NASA-AMMOS/aerie/pull/1513/commits/70b373f98955d4df557f2ae1aac8aac7a92eb0f4)).